controllers/capabilities/tenant_status_reconciler.go

```diff
@@ -0,0 +1,100 @@
+package controllers
+
+import (
+	"context"
+
+	capabilitiesv1alpha1 "github.com/3scale/3scale-operator/apis/capabilities/v1alpha1"
+	"github.com/3scale/3scale-operator/pkg/reconcilers"
+
+	"github.com/3scale/3scale-operator/pkg/apispkg/common"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type TenantStatusReconciler struct {
+	*reconcilers.BaseReconciler
+	resourceUpdated  *capabilitiesv1alpha1.Tenant
+	resourceOriginal *capabilitiesv1alpha1.Tenant
+	reconcileError   error
+	logger           logr.Logger
+}
+
+func NewTenantStatusReconciler(b *reconcilers.BaseReconciler, resourceUpdated, resourceOriginal *capabilitiesv1alpha1.Tenant, reconcileError error) *TenantStatusReconciler {
+	return &TenantStatusReconciler{
+		BaseReconciler:   b,
+		resourceUpdated:  resourceUpdated,
+		resourceOriginal: resourceOriginal,
+		reconcileError:   reconcileError,
+		logger:           b.Logger().WithValues("Status Reconciler", resourceUpdated.Name),
+	}
+}
+
+// Compare and update the status of the tenant CR (both, object and object.status)
+func (s *TenantStatusReconciler) Reconcile() (reconcile.Result, error) {
```

Comment:
Idea here is to have an update to the CR done in one place and one place only to avoid:
1) multiple reconcile loops triggering
2) avoid k8s errors based on resource out of sync

Reply:
Updated